netlify/netlify-plugin-nextjs
# feat: don't move files to CDN if they match middleware

### Summary
When static file moving is enabled, pre-rendered pages are moved to the CDN rather than being served from the origin with next/server. This causes a problem if the site uses middleware, as it does not run on pages served from the CDN. This PR checks any moved files to see if they match any middleware. If they do, it does not move them to the CDN and instead serves them from the origin. It also logs a warning about this.

### Test plan

- Deploy the site with `EXPERIMENTAL_MOVE_STATIC_PAGES`.
- Load `/middle`
- Confirm that the message says middleware is enabled

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

Fixes #802

### Standard checks:





---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.
